tests/lib/workflows/test_send_errored_vm_email.py

```diff
@@ -0,0 +1,589 @@
+    mock_find_user_info,
+    mock_find_servers,
+):
+    """
+    Tests send_errored_vm_email() function actually sends email - as_html True
+    """
+    limit_by_projects = ["project1", "project2"]
+    all_projects = False
+    cloud_account = NonCallableMock()
+    smtp_account = NonCallableMock()
+    mock_kwargs = {"arg1": "val1", "arg2": "val2"}
+
+    mock_query = mock_find_servers.return_value
+
+    # doesn't matter what the values are here since we're just getting the keys
+    mock_query.to_props.return_value = {
+        "user_id1": [],
+        "user_id2": [],
+    }
+    mock_find_user_info.side_effect = [
+        ("user1", "user_email1"),
+        ("user2", "user_email2"),
+    ]
+
+    send_errored_vm_email(
+        smtp_account=smtp_account,
+        cloud_account=cloud_account,
+        time_variable= -1,
+        limit_by_projects=limit_by_projects,
+        all_projects=all_projects,
+        as_html=True,
+        send_email=True,
+        use_override=False,
+        **mock_kwargs,
+    )
+
+    mock_find_servers.assert_called_once_with(cloud_account, -1, limit_by_projects)
+    mock_query.to_props.assert_called_once()
+    mock_find_user_info.assert_has_calls(
+        [
+            call("user_id1", cloud_account, "cloud-support@stfc.ac.uk"),
+            call("user_id2", cloud_account, "cloud-support@stfc.ac.uk"),
+        ]
+    )
+
+    mock_build_email_params.assert_has_calls(
+        [
+            call(
+                "user1",
+                mock_query.to_html.return_value,
+                email_to=["user_email1"],
+                as_html=True,
+                email_cc=None,
+                **mock_kwargs,
+            ),
+            call(
+                "user2",
+                mock_query.to_html.return_value,
+                email_to=["user_email2"],
+                as_html=True,
+                email_cc=None,
+                **mock_kwargs,
+            ),
+        ]
+    )
+
+    mock_query.to_html.assert_has_calls(
+        [
+            call(groups=["user_id1"], include_group_titles=False),
+            call(groups=["user_id2"], include_group_titles=False),
+        ]
+    )
+
+    mock_emailer.assert_has_calls(
+        [
+            call(smtp_account),
+            call().send_emails([mock_build_email_params.return_value]),
+            call(smtp_account),
+            call().send_emails([mock_build_email_params.return_value]),
+        ]
+    )
+
+
+# pylint:disable=too-many-arguments
+@patch("workflows.send_errored_vm_email.find_servers_with_errored_vms")
+@patch("workflows.send_errored_vm_email.find_user_info")
+@patch("workflows.send_errored_vm_email.print_email_params")
+def test_send_errored_vm_email_print(
+    mock_print_email_params,
+    mock_find_user_info,
+    mock_find_servers,
+):
+    """
+    Tests send_errored_vm_email() function prints when send_email=False
+    """
+    limit_by_projects = ["project1", "project2"]
+    all_projects = False
+    time_variable = -1
```

Comment:
you can remove this variable as you're not using it - linting is failing because of this